nish-kishore/sirfunctions
# ❇️ functions for updating local datasets and checking columns where duplicates with the same unique identifier differ

The following pull request addresses #97 and introduces one helper function:
1. `update_polio_data()` updates a local copy of an .Rds object outputted by `get_all_polio_data()`. It takes two parameters: `local_dataset` and `overwrite`. 
2. `get_diff_cols()` checks for where duplicates of the same ID differ. This is helpful in cases where there are records with the same ID. It takes two parameters: `df` and `id_col`.

To test `update_polio_data()`:
1. Run `get_all_polio_data()` and save the output to the local drive.
2. Run `update_polio_data()` passing the path as an argument. Set overwrite to `FALSE`.
3. Run `update_polio_data()` again, passing the path as an argument only. It should overwrite the target dataset.

To test `get_diff_cols()`:
1. Run the following lines:
```
raw.data <- get_all_polio_data(attach.spatial.data = F)
duplicate_epids <- raw.data$afp |> group_by(epid) |> summarise(n=n()) |> filter(n > 1) |> pull(epid)
check_duplicates <- raw.data$afp |> filter(epid %in% duplicate_epids) |> get_diff_cols(id_col = "epid")
```
